File: scripts/apply_global_theme.py
import os
import glob
from bs4 import BeautifulSoup
import re

workspace = r"c:\Users\harsh\OneDrive\Desktop\site"
html_files = glob.glob(os.path.join(workspace, "*.html"))

# 1. Read index.html to extract components
index_path = os.path.join(workspace, "index.html")
with open(index_path, 'r', encoding='utf-8') as f:
    index_soup = BeautifulSoup(f, 'html.parser')

premium_navbar = index_soup.find('nav', class_='premium-navbar')
mobile_nav = index_soup.find('div', class_='premium-mobile-nav')
footer = index_soup.find('footer', class_='footer')

if not premium_navbar or not mobile_nav or not footer:
    print("Error: Could not extract premium navbar, mobile nav, or footer from index.html")
    exit(1)

premium_navbar_str = str(premium_navbar)
mobile_nav_str = str(mobile_nav)
footer_str = str(footer)

for file_path in html_files:
    if os.path.basename(file_path) == "index.html":
        continue # Skip index.html itself

    with open(file_path, 'r', encoding='utf-8') as f:
        content = f.read()
    
    soup = BeautifulSoup(content, 'html.parser')
    
    changed = False

    # 1. Replace Navigation
    existing_nav = soup.find('nav')
    if existing_nav:
        existing_nav.replace_with(BeautifulSoup(premium_navbar_str, 'html.parser'))
        changed = True
    else:
        # Insert after body start
        if soup.body:
            soup.body.insert(0, BeautifulSoup(premium_navbar_str, 'html.parser'))
            changed = True
            
    # 2. Replace Mobile Nav
    existing_mobile = soup.find('div', class_=re.compile(r'.*mobile.*nav.*'))
    if existing_mobile:
        existing_mobile.replace_with(BeautifulSoup(mobile_nav_str, 'html.parser'))
        changed = True
    else:
        # Insert after premium-navbar
        inserted_nav = soup.find('nav', class_='premium-navbar')
        if inserted_nav:
            inserted_nav.insert_after(BeautifulSoup(mobile_nav_str, 'html.parser'))
            changed = True

    # 3. Replace Footer
    existing_footer = soup.find('footer')
    if existing_footer:
        existing_footer.replace_with(BeautifulSoup(footer_str, 'html.parser'))
        changed = True
    else:
        if soup.body:
            soup.body.append(BeautifulSoup(footer_str, 'html.parser'))
            changed = True

    # 4. Ensure Stylesheet
    if soup.head:
        if not soup.head.find('link', href='navbar-premium.css'):
            link_tag = soup.new_tag('link', rel='stylesheet', href='navbar-premium.css')
            soup.head.append(link_tag)
            changed = True
            
    # 5. Ensure JS
    if soup.body:
        if not soup.body.find('script', src='navbar-premium.js'):
            script_tag = soup.new_tag('script', src='navbar-premium.js', defer=True)
            soup.body.append(script_tag)
            changed = True

    if changed:
        with open(file_path, 'w', encoding='utf-8') as f:
            # Prevent formatter from adding closing tags to self-closing HTML5 elements
            f.write(soup.encode(formatter='html5').decode('utf-8'))
        print(f"Updated {os.path.basename(file_path)}")

print("Global theme structure applied successfully.")

File: scripts/apply_global_theme.js
const fs = require('fs');
const path = require('path');
const cheerio = require('cheerio');

const workspace = 'c:\\Users\\harsh\\OneDrive\\Desktop\\site';

async function run() {
    const files = fs.readdirSync(workspace).filter(f => f.endsWith('.html'));

    // Read index.html
    const indexContent = fs.readFileSync(path.join(workspace, 'index.html'), 'utf8');
    const $index = cheerio.load(indexContent);
    
    const premiumNavbar = $index('nav.premium-navbar').parent().html() ? $index('nav.premium-navbar').prop('outerHTML') : '';
    const mobileNav = $index('div.premium-mobile-nav').prop('outerHTML');
    const footer = $index('footer.footer').prop('outerHTML');

    if (!premiumNavbar || !mobileNav || !footer) {
        console.error("Could not extract nav or footer from index.html");
        process.exit(1);
    }

    for (const file of files) {
        if (file === 'index.html') continue;
        
        const filePath = path.join(workspace, file);
        const content = fs.readFileSync(filePath, 'utf8');
        const $ = cheerio.load(content);
        
        let changed = false;

        // 1. Remove light theme media query preference for pure dark mode
        // For inline styles or script logic (mostly handled in CSS, but just to be sure we remove the light class if any)
        if ($('body').hasClass('light-theme')) {
            $('body').removeClass('light-theme');
            changed = true;
        }

        // 2. Replace Nav
        if ($('nav').length) {
            $('nav').first().replaceWith(premiumNavbar);
            changed = true;
        } else {
            $('body').prepend(premiumNavbar);
            changed = true;
        }

        // 3. Replace Mobile Nav
        if ($('div[class*="mobile-nav"]').length) {
            $('div[class*="mobile-nav"]').first().replaceWith(mobileNav);
            changed = true;
        } else {
            $('nav.premium-navbar').after(mobileNav);
            changed = true;
        }

        // 4. Replace Footer
        if ($('footer').length) {
            $('footer').last().replaceWith(footer);
            changed = true;
        } else {
            $('body').append(footer);
            changed = true;
        }

        // 5. CSS and JS dependencies
        if (!$('link[href="navbar-premium.css"]').length) {
            $('head').append('\n  <link rel="stylesheet" href="navbar-premium.css">');
            changed = true;
        }
        
        if (!$('script[src="navbar-premium.js"]').length) {
            $('body').append('\n  <script src="navbar-premium.js" defer></script>');
            changed = true;
        }

        if (changed) {
            fs.writeFileSync(filePath, $.html(), 'utf8');
            console.log(`Updated ${file}`);
        }
    }
    console.log("All pages successfully updated to global layout.");
}

run();
